### PR TITLE
feat: Adds `contacts.checkExistence` endpoint

### PR DESCRIPTION
**Pull request by matheusbsilva137**
## Proposed changes (including videos or screenshots)

 - Added a new `contacts.checkExistence` endpoint, which allows identifying whether there's already a contact registered using a given email, phone, id or visitor to source association.

## Steps to test or reproduce

**Example request (cURL):**
```
curl --request GET \
  --url 'http://localhost:3000/api/v1/omnichannel/contacts.checkExistence?email=teste%40example.com' \
  --header 'X-Auth-Token: jVrwkAmobgfne3aEdL-dJh-znNXMkhg79pEAZS2rEIg' \
  --header 'X-User-Id: rMq9ovRmZabXHW4Do' \
```

**Example response:** `{ exists: true, success: true }`

This endpoints only fails in case invalid query params (or more than one valid param) are provided. Accepted params are: `contactId`, `email`, `phone` and `visitor` (which is an object containing a `visitorId` and a `source`, same as in the `contacts.get` endpoint).

The `view-livechat-contact` permission is **required** in order to use this endpoint.

## Further comments

[CORE-888]

[CORE-888]: https://rocketchat.atlassian.net/browse/CORE-888?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/meteor/tests/end-to-end/api/livechat/contacts.ts [hunk 1]
```diff
@@ -950,6 +950,150 @@ describe('LIVECHAT - contacts', () => {
...
+			await updatePermission('view-livechat-contact', ['admin']);
+			const { body } = await request
+				.post(api('omnichannel/contacts'))
+				.set(credentials)
+				.send({ ...contact });
+			contactId = body.contactId;
+
+			const visitor = await createVisitor(undefined, contact.name, email, phone);
+
+			const room = await createLivechatRoom(visitor.token);
```
Comment (MarcosSpessatto) [inline apps/meteor/tests/end-to-end/api/livechat/contacts.ts:977]: Is it possible to at least close the room after all?

File: apps/meteor/tests/end-to-end/api/livechat/contacts.ts [hunk 2]
```diff
@@ -950,6 +950,150 @@ describe('LIVECHAT - contacts', () => {
...
+		const contact = {
+			name: faker.person.fullName(),
+			emails: [email],
+			phones: [phone],
+			contactManager: agentUser?._id,
+		};
+
+		before(async () => {
+			await updatePermission('view-livechat-contact', ['admin']);
+			const { body } = await request
```
Comment (MarcosSpessatto) [inline apps/meteor/tests/end-to-end/api/livechat/contacts.ts:969]: Not sure if its possible to delete the contact after all, but if possible...

File: packages/rest-typings/src/v1/omnichannel.ts [hunk 3]
```diff
@@ -1418,6 +1418,66 @@ const GETOmnichannelContactsSearchSchema = {
...
+				},
+			},
+			required: ['contactId'],
+			additionalProperties: false,
+		},
+		{
+			type: 'object',
+			properties: {
+				email: {
+					type: 'string',
```
Comment (MarcosSpessatto) [inline packages/rest-typings/src/v1/omnichannel.ts:1446]: Is it possible to ensure this indeed follows an email format?

**Review by MarcosSpessatto — CHANGES_REQUESTED** [review]

File: apps/meteor/app/livechat/server/api/v1/contact.ts [hunk 4]
```diff
@@ -166,6 +167,25 @@ API.v1.addRoute(
...
+	{
+		async get() {
+			const { contactId, visitor, email, phone } = this.queryParams;
+			const filter = {
+				...(email && { 'emails.address': email }),
+				...(phone && { 'phones.phoneNumber': phone }),
+				...(contactId && { _id: contactId }),
+			};
+
+			const contact = await (visitor ? getContactByChannel(visitor) : LivechatContacts.countDocuments(filter));
```
Comment (KevLehman) [inline apps/meteor/app/livechat/server/api/v1/contact.ts:182]: The `count` query should be on the model 👀

File: packages/rest-typings/src/v1/omnichannel.ts (hunk 3 above)
Comment (matheusbsilva137) [reply packages/rest-typings/src/v1/omnichannel.ts:1446]: Done!

File: apps/meteor/tests/end-to-end/api/livechat/contacts.ts (hunk 1 above)
Comment (matheusbsilva137) [reply apps/meteor/tests/end-to-end/api/livechat/contacts.ts:978]: Done!

File: apps/meteor/app/livechat/server/api/v1/contact.ts [hunk 5]
```diff
@@ -166,6 +167,25 @@ API.v1.addRoute(
 	},
 );
 
+API.v1.addRoute(
+	'omnichannel/contacts.checkExistence',
+	{ authRequired: true, permissionsRequired: ['view-livechat-contact'], validateParams: isGETOmnichannelContactsCheckExistenceProps },
+	{
+		async get() {
+			const { contactId, visitor, email, phone } = this.queryParams;
+			const filter = {
```
Comment (KevLehman) [inline apps/meteor/app/livechat/server/api/v1/contact.ts:176]: Just making sure we have a proper index that covers at least the base usecase of the endpoint 👀 
do we?

**Review by KevLehman — CHANGES_REQUESTED** [review]

**Review by KevLehman — DISMISSED** [review]

**Review by MarcosSpessatto — APPROVED** [review]

**Review by KevLehman — APPROVED** [review]

**Review by MarcosSpessatto — APPROVED** [review]

**Pull request merged**